resources/tools/check-markdown/src/check-link.ts

```diff
@@ -0,0 +1,59 @@
+import * as fs from 'node:fs';
+import { posix as path } from 'node:path';
+
+import { Tokens } from 'marked';
+
+import { LinkRef, LinkRefMessage } from './types.js';
+
+export function checkLinks(root: string, links: LinkRef[]) {
+  let result = true;
+  for(const file of links) {
+    for(const link of file.links) {
+      // verify that the file exists in filesystem at expected location
+      result = checkLink(root, file.file, <Tokens.Link>link, file.messages) && result;
+    }
+  }
+  return result;
+}
+
+function checkLink(root: string, file: string, token: Tokens.Link | Tokens.Image, messages: LinkRefMessage[]) {
+  const parsed = token.href.split('#');
+  const href = parsed[0];
+  // const anchor = parsed.length > 1 ? parsed[1] : '';
+
+  if(href.startsWith('https:') || href.startsWith('http:')) {
```

Comment:
A comment somewhere what we check and which links are valid or invalid might be helpful, either in this file or in a  README.